src/lazy/index.tsx

```diff
@@ -1,48 +1,25 @@
-import type { ComponentProps, ComponentType, FunctionComponent } from 'react';
+import { ComponentProps, ComponentType, FunctionComponent } from 'react';
 
 import { PHASE } from '../constants';
 import { hash, displayNameFromId, isNodeEnvironment } from '../utils';
-import type { Asset, Manifest } from '../webpack';
+import { Asset, Manifest } from '../webpack';
 
 import { createComponentClient } from './components/client';
 import { createComponentServer } from './components/server';
 import { createDeferred } from './deferred';
 import { ClientLoader, Loader, ServerLoader } from './loader';
+import { LazyOptions, LazyComponent } from './types';
 
-export type { Asset, Manifest };
-
-export type Options = {
-  /**
-   * Whenever it should be required and rendered in SSR
-   * If false it will just render the provided fallback
-   */
-  ssr?: boolean;
-  /**
-   * Id of `PHASE` responsible for start loading
-   */
-  defer?: number;
-  /**
-   * Id of the module being imported (normally its path).
-   * It's calculated and provided by babel plugin
-   */
-  moduleId?: string;
-};
-
-export type LazyComponent<C extends ComponentType<any>> = FunctionComponent<
-  ComponentProps<C>
-> & {
-  preload: () => void;
-  getAssetUrls: (manifest: Manifest) => string[] | undefined;
-};
+export { Asset, Manifest, LazyOptions, LazyComponent };
 
 function lazyProxy<C extends ComponentType<any>>(
   loader: Loader<C>,
-  { defer = PHASE.PAINT, moduleId = '', ssr = true }: Options = {}
+  { defer = PHASE.PAINT, moduleId = '', ssr = true }: LazyOptions = {}
 ): LazyComponent<C> {
   const isServer = isNodeEnvironment();
   const dataLazyId = hash(moduleId);
 
-  const LazyComponent: FunctionComponent<ComponentProps<C>> = isServer
+  const LazyComponent_: FunctionComponent<ComponentProps<C>> = isServer
```

Comment:
I'm happy to rename this but really don't like the underscore ending. Can we do `LazyBase` maybe? Or `LazyInternal`?

Reply:
Ah yeah I quickly push in a rename.

Reply:
Changes pushed. `LazyComponent_` is now `LazyInternal`.